src/main/java/de/danielbechler/diff/access/Instances.java

```diff
@@ -225,8 +228,10 @@ else if (sourceAccessorType != null)
 				// special handling for beans and arrays should go here
 			}
 		}
-		throw new IllegalArgumentException("Detected instances of different types " + types + ". " +
-				"Instances must either be null or have the exact same type.");
+
+		logger.info("Detected instances of different types " + types + ". " +
+			    "Instances should normally either be null or have the exact same type.");
```

Comment:
So far so good. Maybe replace "Instances should normally either be null or have the exact same type" with something along the lines of "These objects will only be compared via equals method".